processes/ingest/doab.py

```diff
@@ -30,145 +25,67 @@ class DOABProcess(CoreProcess):
-        self.ingestLimit = (int(args[4]) + self.ingestOffset) if args[4] else 10000
-
-        self.generateEngine()
-        self.createSession()
-
         self.s3_manager = S3Manager()
         self.s3_manager.createS3Client()
-        self.s3Bucket = os.environ['FILE_BUCKET']
+        self.s3_bucket = os.environ['FILE_BUCKET']
 
-        self.fileQueue = os.environ['FILE_QUEUE']
-        self.fileRoute = os.environ['FILE_ROUTING_KEY']
+        self.file_queue = os.environ['FILE_QUEUE']
+        self.file_route = os.environ['FILE_ROUTING_KEY']
 
         self.rabbitmq_manager = RabbitMQManager()
         self.rabbitmq_manager.createRabbitConnection()
-        self.rabbitmq_manager.createOrConnectQueue(self.fileQueue, self.fileRoute)
+        self.rabbitmq_manager.createOrConnectQueue(
+            self.file_queue, self.file_route)
+
+        self.offset = int(args[5]) if args[5] else 0
+        self.limit = (int(args[4]) + self.offset) if args[4] else 10000
 
-        self.constants = get_constants()
+        self.dSpace_service = DSpaceService(base_url=self.DOAB_BASE_URL, source_mapping=DOABMapping)
 
     def runProcess(self):
-        if self.process == 'daily':
-            self.importOAIRecords()
-        elif self.process == 'complete':
-            self.importOAIRecords(fullOrPartial=True)
-        elif self.process == 'custom':
-            self.importOAIRecords(startTimestamp=self.ingestPeriod)
-        elif self.process == 'single':
-            self.importSingleOAIRecord(self.singleRecord)
+        try:
+            self.generateEngine()
+            self.createSession()
 
-        self.saveRecords()
-        self.commitChanges()
+            records = []
 
-        logger.info(f'Ingested {len(self.records)} DOAB records')
+            if self.process == 'daily':
+                records = self.dSpace_service.get_records(offset=self.offset, limit=self.limit)
+            elif self.process == 'complete':
+                records = self.dSpace_service.get_records(full_import=True, offset=self.offset, limit=self.limit)
+            elif self.process == 'custom':
+                records = self.dSpace_service.get_records(start_timestamp=self.ingestPeriod, offset=self.offset, limit=self.limit)
+            elif self.singleRecord:
+                record = self.dSpace_service.get_single_record(record_id=self.singleRecord, source_identifier=self.DOAB_IDENTIFIER)
+                self.manage_links(record)
 
-    def parseDOABRecord(self, oaiRec):
-        try:
-            doabRec = DOABMapping(oaiRec, self.OAI_NAMESPACES, self.constants)
-            doabRec.applyMapping()
-        except MappingError as e:
-            raise DOABError(e.message)
+            if records:
+                for record in records:
+                    self.manage_links(record)
+
+            self.saveRecords()
+            self.commitChanges()
+
+            logger.info(f'Ingested {len(self.records) if records else 1} DOAB records')
```

Comment:
Shouldn't the result of the else conditional be 0 if the records array is empty?

Reply:
Yes, this should still return 0 since the len of the records array would be 0 if it is empty.